packages/artifact/src/internal/upload/blob-upload.ts

```diff
@@ -41,13 +48,29 @@ export async function uploadZipToBlobStorage(
     onProgress: uploadCallback
   }
 
+  let md5Hash: string | undefined = undefined
+  const uploadStream = new stream.PassThrough()
+  const hashStream = crypto.createHash('md5')
+
+  zipUploadStream.pipe(uploadStream) // This stream is used for the upload
+  zipUploadStream.pipe(hashStream).setEncoding('hex') // This stream is used to compute a hash of the zip content that gets used. Integrity check
+
   try {
+    core.info('Beginning upload of artifact content to blob storage')
+
     await blockBlobClient.uploadStream(
-      zipUploadStream,
+      uploadStream,
       bufferSize,
       maxBuffers,
       options
     )
+
+    core.info('Finished uploading artifact content to blob storage!')
+
+    hashStream.end()
+    md5Hash = hashStream.read() as string
+    core.info(`MD5 hash of uploaded artifact zip is ${md5Hash}`)
```

Comment:
Out of curiosity, will printing the hash get a little noisy? Maybe we could make this a debug log?

Reply:
I'd prefer to at last initially keep this in the logs. We're not going to immediately expose the hash via our REST APIs for artifacts unless we do more work so there is no real way for users to get this information unless we log it. When we update expose this potentially in our APIs then we could make this debug